### PR TITLE
[FEATURE REQUEST] Support service account impersonation with Cloud Spanner

### PR DESCRIPTION
**Pull request by Attsun1031**
## Overview
Hello, I am a tbls user, mainly with postgres and Cloud spanner.

We want to use [service account impersonation](https://cloud.google.com/iam/docs/create-short-lived-credentials-direct) for Cloud Spanner to keep granting minimum permissions to each service accounts.

It is not difficult to implement this feature, so I made draft implementation (this PR).
Please review design of this feature.

## Design
Using URL values to pass params about impersonate service account.

New values are below.

- `impersonate_service_account`: An email of service account that is impersonated by caller.
- `impersonate_service_account_duration`: The duration parameter for impersonation request.

### Example
- ServiceAccount A: the account who issues a request to impersonate ServiceAccountB. ServiceAccountA does not have permission to connect spanner directly.
- ServiceAccount B: the account who can connect to spanner databases, and used from ServiceAccountA.

In this case, the command will be below.

```
tbls doc \
  “spanner://{project}/{instance}/{database}?creds={credential path of ServiceAccount A}&impersonate_service_account={Email of ServiceAccount B}&impersonate_service_account_duration={duration of impersonation}”
```

## Next
- If ok, I will implement below
  - unit tests
  - documents
  - support bigquery

### CODE REVIEW

**Comment by k1LoW** [comment]
Thanks for the good suggestion.
It looks like this implementation is not only related to the connection to Spanner, but also to BigQuery.

I will also check other software implementations (URL queries or environment variables, etc.)

**Comment by Attsun1031** [comment]
@k1LoW Thanks! I'll implement to support bigquery after design policy is fixed.

**Comment by k1LoW** [comment]
How about an implementation that accepts the environment variables `GOOGLE_IMPERSONATE_SERVICE_ACCOUNT` and `GOOGLE_IMPERSONATE_SERVICE_ACCOUNT_LIFETIME` ?

`GOOGLE_IMPERSONATE_SERVICE_ACCOUNT` seems to be an officially accepted environment variable.

ref: https://cloud.google.com/blog/topics/developers-practitioners/using-google-cloud-service-account-impersonation-your-terraform-code ( [JA](https://cloud.google.com/blog/ja/topics/developers-practitioners/using-google-cloud-service-account-impersonation-your-terraform-code) )

`GOOGLE_IMPERSONATE_SERVICE_ACCOUNT_LIFETIME` is a non-standard environment variable, but it is easy to understand that it is a value related to `GOOGLE_IMPERSONATE_SERVICE_ACCOUNT`.
The reason I chose `LIFETIME` instead of `DURATION` is that it matches [the actual field name](https://github.com/k1LoW/tbls/pull/360/files#diff-f7d5a7e03c129f542c306c0c6c419484034711d51cd5d8a26fc7e1334cb9fbcdR159).

File: datasource/gcp.go [hunk 1]
```diff
@@ -125,3 +138,24 @@ func setEnvGoogleApplicationCredentials(values url.Values) error {
...
+	impersonateServiceAccount := values.Get("impersonate_service_account")
+	if impersonateServiceAccount == "" {
+		return nil, nil
+	}
+	// Setting up options for service account impersonation
+	durationStr := values.Get("impersonate_service_account_duration")
+	if durationStr == "" {
+		durationStr = "300s"
+	}
+	duration, err := time.ParseDuration(durationStr)
```
Comment (k1LoW) [inline datasource/gcp.go:152]: I suggest using https://github.com/k1LoW/duration to also accept more human readable values

**Comment by Attsun1031** [comment]
@k1LoW Thanks ! It looks good to me.

I'll try to implement using env vars.

**Comment by Attsun1031** [comment]
@k1LoW I fixed the implementation, and added docs and tests. Can you review it?

**Comment by k1LoW** [comment]
Sorry for the delay (I was out sick). I will review as soon as I can.

**Review by k1LoW — APPROVED** [review]
LGTM! Thank you!!

**Pull request merged**